validator/keymanager/v2/derived/derived.go

```diff
@@ -369,20 +369,53 @@ func (dr *Keymanager) FetchValidatingPublicKeys(_ context.Context) ([][48]byte,
 	return result, nil
 }
 
+// FetchValidatingPrivateKeys fetches the list of validating private keys from the keymanager.
+func (dr *Keymanager) FetchValidatingPrivateKeys(ctx context.Context) ([][]byte, error) {
+	dr.lock.RLock()
+	privKeys := make([][]byte, len(dr.secretKeysCache))
+	pubKeys, err := dr.FetchValidatingPublicKeys(ctx)
+	if err != nil {
+		return nil, errors.Wrap(err, "could not retrieve public keys")
+	}
+	for i, pk := range pubKeys {
+		seckey, ok := dr.secretKeysCache[pk]
+		if !ok {
+			return nil, errors.New("Could not fetch private key")
+		}
+		privKeys[i] = seckey.Marshal()
+	}
+	dr.lock.RUnlock()
+	return privKeys, nil
+}
```

Comment:
There are codepaths here that could exit without unlocking this lock. Please defer the unlocking of this lock
```suggestion
func (dr *Keymanager) FetchValidatingPrivateKeys(ctx context.Context) ([][]byte, error) {
	dr.lock.RLock()
	defer dr.lock.RUnlock()
	privKeys := make([][]byte, len(dr.secretKeysCache))
	pubKeys, err := dr.FetchValidatingPublicKeys(ctx)
	if err != nil {
		return nil, errors.Wrap(err, "could not retrieve public keys")
	}
	for i, pk := range pubKeys {
		seckey, ok := dr.secretKeysCache[pk]
		if !ok {
			return nil, errors.New("Could not fetch private key")
		}
		privKeys[i] = seckey.Marshal()
	}
	return privKeys, nil
}
```